test/global.test.ts

```diff
@@ -93,9 +94,30 @@ export async function createBuildTask(folder: vscode.WorkspaceFolder): Promise<v
 
         allTasks.push(buildTask);
         await workspaceTasks.update('tasks', allTasks, vscode.ConfigurationTarget.WorkspaceFolder);
+        console.log(`${testName}: Build task created`);
+        verifyBuildTask(testName, folder);
     }
 }
 
+function verifyBuildTask(testName: string, folder: vscode.WorkspaceFolder): void {
+    const maxTry = 5;
+    let i = 0;
+    let buildTaskPresent = isBuildTaskPresent(folder);
+
+    while (!buildTaskPresent && i < maxTry) {
+        buildTaskPresent = isBuildTaskPresent(folder);
+        i++;
+    }
+    assert.equal(buildTaskPresent, true);
+    console.log(`${testName}: Verified the build task is present in the workspace after ${i} tries`);
+}
+
+function isBuildTaskPresent(folder: vscode.WorkspaceFolder): boolean {
+    const workspaceTasks = vscode.workspace.getConfiguration('tasks', folder.uri);
+    const allTasks = workspaceTasks && workspaceTasks.tasks as TaskDefinitionBase[] || [];
+    return allTasks.findIndex(t => t.label === 'build') !== -1
```

Comment:
```suggestion
    return allTasks.some(t => t.label === 'build');
```
`some` is basically the same as C#'s `Any`